src/app/modules/item/services/item-datasource.service.ts

```diff
@@ -68,21 +70,54 @@ export class ItemDataSource implements OnDestroy {
   private fetchItemData(navItem: NavItem): Observable<ItemData> {
     return forkJoin([
       this.getBreadcrumb(navItem),
-      this.getItemByIdService.get(navItem.itemId).pipe(
-        switchMap(item => this.startResultIfNeeded(navItem, item).pipe(map(a => ({ item: item, attemptId: a }))))
-      )
+      this.getItemByIdService.get(navItem.itemId)
     ]).pipe(
-      map(([breadcrumbs, {item: i, attemptId: a}]) => ({ item: i, breadcrumbs: breadcrumbs, attemptId: a, nav: navItem })),
+      switchMap(([breadcrumbs, item]) => {
+        // emit immediately without result, then fetch and add it
+        const initialData = { nav: navItem, item: item, breadcrumbs: breadcrumbs };
+        return concat(
+          of(initialData),
+          this.fetchResults(navItem, item).pipe(
+            map(r => ({ ...initialData, ...r}))
+          )
+        );
+      })
     );
   }
 
-  private startResultIfNeeded(navItem: NavItem, item: Item): Observable<string|null> { // observable of the result attempt_id
-    const attemptId  = navItem.attemptId || navItem.parentAttemptId;
-    // if not allowed to start a result on this attempt, do not try
-    if (item.requires_explicit_entry || !canCurrentUserViewItemContent(item) || !attemptId) return of(null);
-    return this.resultActionsService.start(navItem.itemPath.concat([navItem.itemId]), attemptId).pipe(
-      mapTo(attemptId),
-      catchError(_e => of(null)) // if got an error, continue with no result
+  private fetchResults(nav: NavItem, item: Item): Observable<{ results: Result[], currentResult?: Result }> {
+    let attempt: { attemptId: string } | { parentAttemptId: string };
+    if (nav.parentAttemptId) attempt = { parentAttemptId: nav.parentAttemptId };
+    else if (nav.attemptId) attempt = { attemptId: nav.attemptId };
+    else return EMPTY; // unexpected
+
+    return this.getResultsService.getResults(nav.itemId, attempt).pipe(
+      switchMap(results => {
+        // 1) if attempt_id was given as arg, try to select the matching result
+        if (nav.attemptId) {
+          const currentResult = results.find(r => r.attemptId === nav.attemptId);
+          if (currentResult) return of({ results: results, currentResult: currentResult});
+        }
+        // 2) if there are already results on this item, select the best one
+        const currentResult = bestAttemptFromResults(results);
+        if (currentResult !== null) return of({ results: results, currentResult: currentResult});
```

Comment:
```suggestion
        if (currentResult) return of({ results: results, currentResult: currentResult});
```

Reply:
For the moment, "everywhere" in the code, nulls are checked if `=== null` and `!== null`. I should recheck if it is really appropriate to do this, but at least for this PR it is consistent with the rest of the code.